### ISSUE TITLE
Module not found: Error: Can't resolve 'nanogallery2'

### ISSUE THREAD

**Issue by khacbinh** [nanostudio-org/nanogallery2#252]
I'm using Vue with vue-cli.
I've installed nanogallery2 by running `npm install nanogallery2` and then import it with `import 'nanogallery2'` in main.js file but it thrown this error
`Failed to compile.`
`Module not found: Error: Can't resolve 'nanogallery2'`
I also tried
`import 'nanogallery2/dist/jquery.nanogallery2.js'`
`require('nanogallery2')`
`require('nanogallery2/dist/jquery.nanogallery2.js')`
but all failed.
Anyone have any idea to make it work?

**Comment by seahindeniz** [comment]
Hi @Kris-B 

There are some missing entries in the package.json file.
After you setup, the issue will be resolved.

First, you need to specify `main` field to let compilers to recognize the main script file.
https://docs.npmjs.com/files/package.json#main
`package.json`
```js
{
  "version": "3.0.2",
  "main": "./src/jquery.nanogallery2.js",
  ...
}
```

This step will resolve this issue. Users just need to import the nanogallery2 module by using one of these ways:
`fileA.js`
```js
// CommonJS
require('nanogallery2');

// ES6
import 'nanogallery2';
```

For the type support, you need to set the `types` field to point the declaration file that you probably need to create one.
https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

`jquery.nanogallery2.d.ts`
```ts
interface JQuery {
  nanogallery2(): JQuery;
  nanogallery2(options: { option1: boolean }): JQuery;
  nanogallery2(api: string): JQuery;
}
```

Then, set the necessary field on
`package.json`
```js
{
  "version": "3.0.2",
  "main": "./src/jquery.nanogallery2.js",
  "types": "./jquery.nanogallery2.d.ts",
  ...
}
```

I would create a PR for the type definitions but I kinda don't know projects API structure and what are the overloads it has.

**Comment by Kris-B** [comment]
Hi @seahindeniz 

Many thanks for your help. I understand that the `main` entry in the `package.json` file will fix the issue opened by @khacbinh, but I'm not sure to understand the second part of your answer. Why is `types` required ?

**Comment by seahindeniz** [comment]
The second part was just a suggestion. It will be useful for JS and TS users to use this library with IntelliSense support. It's not required, of course, I should state the second part is optional. My mistake, sorry 😅

**Comment by Kris-B** [comment]
Ok, that's now more clear to me. Many thanks for your contribution!

**Comment by Kris-B** [comment]
npmjs issue fixed in v3.0.3
https://github.com/nanostudio-org/nanogallery2/releases/tag/v3.0.3
